package.json

```diff
@@ -1,25 +1,28 @@
 {
-    "name": "storybook-builder-vite-parent",
-    "version": "0.1.13",
-    "description": "An experimental plugin to run and build Storybooks with Vite",
-    "main": "index.js",
-    "author": "",
-    "license": "MIT",
-    "scripts": {
-        "lint": "prettier --check .",
-        "format": "prettier --write ."
-    },
-    "workspaces": [
-        "packages/example-react",
-        "packages/example-svelte",
-        "packages/example-vue",
-        "packages/example-workspaces",
-        "packages/storybook-builder-vite"
-    ],
-    "prettier": {
-        "singleQuote": true
-    },
-    "devDependencies": {
-        "prettier": "^2.3.2"
-    }
+  "name": "storybook-builder-vite-parent",
+  "version": "0.1.13",
+  "description": "An experimental plugin to run and build Storybooks with Vite",
+  "license": "MIT",
+  "engines": {
+    "node": ">=16.0.0"
+  },
+  "scripts": {
+    "lint": "eslint \"packages/*/**/*.{js,jsx,mjs,cjs}\" --fix",
+    "lint-ci": "eslint \"packages/*/**/*.{js,jsx,mjs,cjs}\""
```

Comment:
I think we still need a scripts to run prettier to check that the formatting is correct.  I see you have eslint set up to disable the style rules using https://github.com/prettier/eslint-config-prettier, but aside from prettier running in editors, there's nothing that will enforce a correct style in CI, as far as I can tell.

Reply:
We have ESLint, It works in IDEs or Terminal/Command line.

Can you please explain, why you need the Prettier for check formatting?

> enforce a correct style in CI

I think, you doing something wrong, because in CI stage you only check code.

Can you provide an example of enforcing a correct style in CI?

Also, right now we have zero style checks in CI (GIthub Actions).

Reply:
I believe the main point of using prettier is to enforce a consistent code style.  If we are not checking it in CI (which you're right, we aren't right now, but we should), then someone can commit a change without running prettier, and then someone else in another PR will run prettier and change lines that are not related to their PR.  By enforcing that each PR that is merged is correctly formatted, we can guarantee that prettier will not change unrelated lines.

Reply:
We can avoid it by:

* using ESLint before commit changes:

  ```
  yarn lint
  ```
* using ESLint in CI stage

  ```
  yarn lint-ci
  ```

I already done it.

Reply:
`yarn lint` only runs eslint.  ESLint is configured to not enforce style rules, deferring to prettier.  How is prettier formatting done as a part of this?  The config includes `eslint-config-prettier`, which `Turns off all rules that are unnecessary or might conflict with Prettier.`

I see you added `eslint-plugin-prettier` to the devDependencies, but it is not being used in the config.  Nor do I think it should be.  I am of the opinion that prettier should be run on its own, not via eslint, as recommended in https://prettier.io/docs/en/integrating-with-linters.html.

Reply:
> I see you added `eslint-plugin-prettier` to the devDependencies, but it is not being used in the config.

You are wrong.

I do it in `extends`:

https://github.com/eirslett/storybook-builder-vite/blob/22c3b98c9b9bb935cfb8d76de8d306ce0cc21607/.eslintrc.js#L10

Reply:
We could do something like
```
"lint-ci": "eslint \"packages/*/**/*.{js,jsx,mjs,cjs}\" && prettier --check ."
``` 

which would first run eslint, and then prettier. But it could be redundant (time-wasting) - the eslint preset for prettier is supposed to run pretty much the exact same rules that `prettier --check` does. There could be discrepancies due to differences in implementation between eslint and prettier, but that's more of a theoretical concern, I believe. 🤷

Reply:
> We could do something like

@eirslett it's already done by:

https://github.com/eirslett/storybook-builder-vite/blob/49ac41cccf5dbd74be99c761bf2afad7d0a659ee/package.json#L11

> But it could be redundant (time-wasting) - the eslint preset for prettier is supposed to run pretty much the exact same rules that prettier --check does.

Yes, you are right.

Reply:
> the eslint preset for prettier is supposed to run pretty much the exact same rules that prettier --check does.

Unfortunately this doesn't seem to be the case, see https://github.com/eirslett/storybook-builder-vite/pull/193#issuecomment-1006205181.

Reply:
We can add other plugins later.

Current set of plugins is working great for the `storybook-builder-vite` package.